### ISSUE TITLE
NoMethodError: undefined method `empty?' for nil:NilClass

### ISSUE THREAD

**Issue by xgrei** [alaz/legitbot#28]
Thanks for this gem, I have seen a lot of error logs recently like this:

```
NoMethodError: undefined method `empty?' for nil:NilClass
…ms/legitbot-1.0.0/lib/legitbot/validators/ip_ranges.rb:   43:in `valid_ip?'
…ms/legitbot-1.0.0/lib/legitbot/validators/ip_ranges.rb:   20:in `valid_ip?'
```

I think it is caused by the strange ip of the visitor. Can I handle the null value?

**Comment by alaz** [comment]
Hello,

Do you know the IP causing this?

**Comment by alaz** [comment]
Never mind, I can reproduce this

```ruby
Alaz-Laptop:~ alaz$ irb -r interval_tree
Ignoring bson-4.7.1 because its extensions are not built. Try: gem pristine bson --version 4.7.1
irb(main):001:0> t = IntervalTree::Tree.new([])
irb(main):002:0> t.search(1)
=> nil
irb(main):003:0> t.search(2)
=> nil
irb(main):004:0> t = IntervalTree::Tree.new([(0...3)])
irb(main):005:0> t.search(2)
=> [0...3]
irb(main):006:0> t.search(7)
=> []
```

**Comment by alaz** [comment]
@xgrei just wanted to let you know. I suspect that `whois` returns an empty list of Facebook's AS in your environment. You might want to investigate this further. See the commands here in Facebook documentation: https://developers.facebook.com/docs/sharing/webmasters/crawler

**Comment by xgrei** [comment]
Thanks for reply, I cannot get the problematic ip address,  Can you just change the [code](https://github.com/alaz/legitbot/blob/master/lib/legitbot/validators/ip_ranges.rb#L43) from `!ranges.empty?` to `!(ranges && ranges&.empty?`

**Comment by alaz** [comment]
Could you please show the rest of stack trace?

**Comment by xgrei** [comment]
```
NoMethodError: undefined method `empty?' for nil:NilClass
…ms/legitbot-1.0.0/lib/legitbot/validators/ip_ranges.rb:   43:in `valid_ip?'
…ms/legitbot-1.0.0/lib/legitbot/validators/ip_ranges.rb:   20:in `valid_ip?'
…uby/2.6.0/gems/legitbot-1.0.0/lib/legitbot/botmatch.rb:   25:in `valid?'
…uby/2.6.0/gems/legitbot-1.0.0/lib/legitbot/legitbot.rb:   26:in `select'
…uby/2.6.0/gems/legitbot-1.0.0/lib/legitbot/legitbot.rb:   26:in `bot'
…200325332231/app/controllers/application_controller.rb:  160:in `check_bot'
```

```
# application_controller.rb
@is_search_bot = ::Legitbot.bot(request.user_agent, request.remote_ip)&.valid? || false
```

**Comment by xgrei** [comment]
client user agent:

```
Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_1) AppleWebKit/601.2.4 (KHTML, like Gecko) Version/9.0.1 Safari/601.2.4 facebookexternalhit/1.1 Facebot Twitterbot/1.0
```

**Comment by alaz** [comment]
Thank you for reporting the bug. The fix has been included in version 1.0.1

**Comment by xgrei** [comment]
Thank you very much for help. :)
